sdk/cosmos/azure-spring-data-cosmos-core/src/main/java/com/azure/spring/data/cosmos/core/ReactiveCosmosTemplate.java

```diff
@@ -370,21 +351,37 @@ public void setApplicationContext(@NonNull ApplicationContext applicationContext
         generateIdIfNullAndAutoGenerationEnabled(objectToSave, domainType);
         final JsonNode originalItem = prepareToPersistAndConvertToItemProperties(objectToSave);
         final CosmosItemRequestOptions options = new CosmosItemRequestOptions();
+        CosmosAsyncContainer cosmosAsyncContainer = cosmosAsyncClient.getDatabase(this.databaseName)
+                                                                     .getContainer(containerName);
+        Mono<CosmosItemResponse<JsonNode>> item;
         if (partitionKey == null) {
-            partitionKey = PartitionKey.NONE;
+            //  if the partition key is null, SDK will get the partitionKey from the object
+            item = cosmosAsyncContainer.createItem(originalItem, options);
+        } else {
+            item = cosmosAsyncContainer.createItem(originalItem, partitionKey, options);
```

Comment:
Can we update logic in CosmosAsyncContainer#createItem

from

```
    public <T> Mono<CosmosItemResponse<T>> createItem(
        T item,
        PartitionKey partitionKey,
        CosmosItemRequestOptions options) {
        if (options == null) {
            options = new CosmosItemRequestOptions();
        }
        ModelBridgeInternal.setPartitionKey(options, partitionKey);
        return createItem(item, options);
    }
```

to 

```
    public <T> Mono<CosmosItemResponse<T>> createItem(
        T item,
        PartitionKey partitionKey,
        CosmosItemRequestOptions options) {
        if (options == null) {
            options = new CosmosItemRequestOptions();
        }
        if (partitionKey != null) {
            ModelBridgeInternal.setPartitionKey(options, partitionKey);
        }
        return createItem(item, options);
    }
```

So we can delete  `if (partitionKey == null)` here.

Reply:
It doesn't matter, as if the partitionKey is null, SDK will compute it based on the logic mentioned below. 
I have kept the scope of this PR to touch minimal code in the Cosmos Java SDK layer.

Reply:
Then can we update code from

```
if (partitionKey == null) {
    //  if the partition key is null, SDK will get the partitionKey from the object
    item = cosmosAsyncContainer.createItem(originalItem, options);
} else {
    item = cosmosAsyncContainer.createItem(originalItem, partitionKey, options);
}
```
to
```
 // it's OK if partitionKey as null
item = cosmosAsyncContainer.createItem(originalItem, partitionKey, options);
```

Reply:
Yes, good suggestion @chenrujun  - I have updated the code.